criu/sk-unix.c

```diff
@@ -2044,13 +2051,18 @@ static int init_unix_sk_info(struct unix_sk_info *ui, UnixSkEntry *ue)
 	INIT_LIST_HEAD(&ui->node);
 	INIT_LIST_HEAD(&ui->scm_fles);
 	INIT_LIST_HEAD(&ui->ghost_node);
+	INIT_HLIST_NODE(&ui->hash);
 
 	return 0;
 }
 
 int unix_prepare_root_shared(void)
 {
 	struct unix_sk_info *ui;
+	int i;
+
+	for (i = 0; i < SK_INFO_HASH_SIZE; i++) /* initialize hashtable */
+		INIT_HLIST_HEAD(&sk_info_hash[i]);
```

Comment:
Looks like you add to the hlist before hlist initialization:

```
cr_restore_tasks
| crtools_prepare_shared
|   prepare_files
|     collect_image(&files_cinfo)
|       collect_one_file
|         collect_one_file_entry(&unix_sk_cinfo)
|           collect_one_unixsk
|             hlist_add_head(&sk_info_hash)
|
+-restore_root_task
    fork_with_pid
      restore_task_with_children
        if (current->parent == NULL) # root task
          root_prepare_shared
            unix_prepare_root_shared
              INIT_HLIST_HEAD(&sk_info_hash)
```

>Initializing hlists in prepare_files looks like a right place on restore, but you miss dump.

You can add new init_sk_info_hash() helper and call it from there.